console2/src/components/organisms/SecretProjectActivity/index.tsx

```diff
@@ -18,93 +18,118 @@
         try {
             setUpdating(true);
-            await apiUpdateSecretProject(orgName, secretName, projectNameValue || '');
-            setSubmittedProjectName(projectNameValue);
+            await apiUpdateSecretProject(orgName, secretName, newProjects.map(project => project.id));
         } catch (e) {
             setError(e);
         } finally {
             setUpdating(false);
         }
-    }, [orgName, secretName, projectNameValue]);
+    }, [orgName, secretName, newProjects]);
 
     const onConfirmHandler = useCallback(async () => {
-        setShowConfirm(false);
-
         await update();
 
+        setShowConfirm(false);
         setDirty(false);
-    }, [update]);
-
+        setEditMode(false);
+        setExistingProjects(newProjects);
+    }, [update, newProjects]);
+ 
     const onCancelHandler = useCallback(() => {
         setShowConfirm(false);
-    }, []);
+        setNewProjects(existingProjects);
+    }, [existingProjects]);
 
     return (
         <>
             {error && <RequestErrorActivity error={error} />}
 
             <Form loading={updating}>
-                <Form.Group widths={3}>
+                {editMode? (
+                    <Form.Group widths={3}>
                     <Form.Field>
                         <ProjectSearch
                             orgName={orgName}
-                            placeholder="any"
+                            placeholder="Search for projects"
                             fluid={true}
-                            defaultProjectName={submittedProjectName}
-                            onReset={(value) => {
-                                setDirty(false);
-                                setProjectNameValue(value?.name);
-                            }}
-                            onClear={() => {
-                                setDirty(true);
-                                setProjectNameValue(undefined);
-                            }}
-                            onSelect={(value) => {
-                                setDirty(true);
-                                setProjectNameValue(value.name);
+                            onSelect={(project) => {
+                                if(!newProjects.map(project => project.id).includes(project.id)){
+                                    setNewProjects(newProjects.concat(project));
+                                }
                             }}
+                            projectsToNotShow={newProjects}
+                            clearOnSelect={true}
                         />
                     </Form.Field>
-
-                    <Form.Button
-                        primary={true}
-                        negative={true}
-                        content="Update"
-                        disabled={!dirty}
-                        onClick={() => setShowConfirm(true)}
-                    />
                 </Form.Group>
-
+                ):null}
+                <Form.Group>
```

Comment:
There's a UI bug where the views are not updated after the projects are updated. The immediate view _is_ updated, but parent and neighbors are not.


https://user-images.githubusercontent.com/4554569/225025941-691aeb09-98d2-4c53-94be-66f56e11c5b7.mov



A page refresh sorts things out, but there's a case where things can be confusing and re-add unexpected projects if the user doesn't refresh. An example
1. remove an existing project from a secret
2. immdate view is updated to reflect correct projects (or none)
3. go to another tab, e.g. `Info` -- `Restricted to a project` element is stale  
4. Return to `Settings` tab -- now the Projects view is stale, showing the old project that was removed
5. Add another project -- now the Secret's Projects have been updated to include the previously removed project plus the new one




https://user-images.githubusercontent.com/4554569/225025156-b28150c2-bea7-4d57-ac39-7fbce28c7463.mov

Reply:
https://user-images.githubusercontent.com/13145963/225324572-a3a9daf4-857b-460d-8bba-40f2bb7b68a5.mov

Thanks Ben for catching this bug. resolved it.